crates/vm/src/arch/integration_api.rs

```diff
@@ -133,6 +136,14 @@ pub trait VmCoreChip<F, I: VmAdapterInterface<F>> {
         reads: I::Reads,
     ) -> Result<(AdapterRuntimeContext<F, I>, Self::Record)>;
 
+    fn execute_instruction2<Mem, Ctx>(
```

Comment:
depends how we want to handle this wrapper pattern, indeed you need something like this similar to `SingleTraceStep`. it might be possible to just directly implement `InsExecutor` on the wrapper struct? anyways we can keep this for now if it make life easier

Reply:
i moved the execute functions into a new trait for now since we're deleting `VmCoreChip`